app/models/manageiq/providers/kubevirt/infra_manager/provision/cloning.rb

```diff
@@ -82,4 +79,111 @@ def start_clone(options)
     # be used to check if it has been already created:
     phase_context[:new_vm_ems_ref] = offline_vm.metadata.uid
   end
+
+  def create_persistent_volume_claims(pvcs, params, namespace)
+    pvcs.each do |pvc|
+      pvc = param_substitution!(pvc, params)
+
+      pvc.deep_merge!(
+        :metadata => {
+          :namespace => namespace
+        }
+      )
+
+      # Send the request to create the persistent volume claim:
+      source.with_provider_connection do |connection|
+        connection.create_pvc(pvc)
+      end
+    end
+  end
+
+  private
+
+  def offline_vm_from_objects(objects)
+    vm = nil
+    objects.each do |object|
+      if object.kind == "OfflineVirtualMachine"
+        vm = to_hash(object)
+      end
+    end
+    # make sure there is one
+    raise MiqException::MiqProvisionError if vm.nil?
+    vm
+  end
+
+  def persistent_volume_claims_from_objects(objects)
+    pvcs = []
+    objects.each do |object|
+      if object.kind == "PersistentVolumeClaim"
+        pvcs << to_hash(object)
+      end
+    end
+    pvcs
+  end
+
+  def values(template, options)
+    default_params = {}
+    template.parameters.each do |param|
+      name = param.name.downcase
+      value = options[name.to_sym]
+      if value && name == "memory"
+        value = value.to_s + 'Mi'
+      end
+
+      default_params[name] = value || param.value
+    end
+    default_params
+  end
+
+  def param_substitution!(object, params)
+    result = object
+    case result
+    when Hash
+      result.each do |k, v|
+        result[k] = param_substitution!(v, params)
+      end
+    when Array
+      result.map { |v| param_substitution!(v, params) }
+    when String
+      result = substitute_param(params, object)
+    end
+    result
+  end
+
+  def substitute_param(params, object)
```

Comment:
we have here ```param_substitution``` and ```substitute_param```, while reading the code, it is clear each responsibility, but naming 'too close', perhaps we need a better way to distinct them.

Reply:
I was never good at names but will try to make sure they are descriptive.